crates/rome_js_analyze/src/semantic_analyzers/nursery/no_useless_this_alias.rs

```diff
@@ -0,0 +1,176 @@
+use crate::{control_flow::AnyJsControlFlowRoot, semantic_services::Semantic, JsRuleAction};
+use rome_analyze::{context::RuleContext, declare_rule, ActionCategory, Rule, RuleDiagnostic};
+use rome_console::markup;
+use rome_diagnostics::Applicability;
+use rome_js_factory::make;
+use rome_js_semantic::ReferencesExtensions;
+use rome_js_syntax::{
+    AnyJsBinding, AnyJsBindingPattern, AnyJsExpression, JsArrowFunctionExpression,
+    JsAssignmentExpression, JsExpressionStatement, JsIdentifierBinding, JsIdentifierExpression,
+    JsThisExpression, JsVariableDeclaration, JsVariableDeclarator, T,
+};
+use rome_rowan::{AstNode, AstSeparatedList, BatchMutationExt};
+
+declare_rule! {
+    /// Disallow useless `this` aliasing.
+    ///
+    /// Arrow functions inherits `this` from their enclosing scope.
+    /// This makes `this` aliasing useless in this situation.
+    ///
+    /// Credits: https://typescript-eslint.io/rules/no-this-alias/
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// class A {
+    ///     method() {
+    ///         const self = this;
+    ///         return () => {
+    ///             return self;
+    ///         }
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// ## Valid
+    ///
+    /// ```js
+    /// class A {
+    ///     method() {
+    ///         const self = this;
+    ///         return function() {
+    ///             this.g();
+    ///             return self;
+    ///         }
+    ///     }
+    /// }
+    /// ```
+    ///
+    pub(crate) NoUselessThisAlias {
+        version: "next",
+        name: "noUselessThisAlias",
+        recommended: true,
+    }
+}
+
+impl Rule for NoUselessThisAlias {
+    type Query = Semantic<JsVariableDeclarator>;
+    type State = JsIdentifierBinding;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let declarator = ctx.query();
+        let model = ctx.model();
+        let mut is_this_alias = false;
+        if let Some(initializer) = declarator.initializer() {
+            let initializer = initializer.expression().ok()?.omit_parentheses();
+            if !JsThisExpression::can_cast(initializer.syntax().kind()) {
+                return None;
+            }
+            is_this_alias = true;
+        };
```

Comment:
```suggestion
        let is_this_alias = if let Some(initializer) = declarator.initializer() {
            let initializer = initializer.expression().ok()?.omit_parentheses();
            if !JsThisExpression::can_cast(initializer.syntax().kind()) {
                return None;
            }
            true
        } else {
        	false
        };
```

Can you do this?

Reply:
You could also compress everything by using `declarator.initializer().map(..).unwrap_or_default`.

Reply:
> You could also compress everything by using
> declarator.initializer().map(..).unwrap_or_default().

This is mostly a style choice?

Reply:
Throughout the code base, we always try to reduce the number of branches if we have methods that allow us to do so.